test/functional/ui/inccommand_spec.lua

```diff
@@ -2291,6 +2292,75 @@ describe(":substitute", function()
     ]])
   end)
 
+  it("inccommand=split, contraction of two subsequent NL chars", function()
+    -- luacheck: push ignore 611
+    local text = [[
+      AAA AA
+      
+      BBB BB
+      
+      CCC CC
+      
+]]
```

Comment:
could avoid lucheck push/pop by:

    local text = 'AAA AA\n\nBBB BB\n\nCCC CC\n\n'

Reply:
There is no reason that luacheck should care about anything that happens between `[[` and `]]`. I would rather try to fix the linter than changing a lot of multiline strings there is nothing wrong with.

Reply:
there aren't a lot of these though.

Reply:
> there aren't a lot of these though.

There is going to be a lot of these when writing extmark/byte-update tests. I do not like needing to completely rewrite a perfectly readable multi-line string because there is a double newline or TAB character (or something else that the linter is not supposed to care about) somewhere in there.

>  Whereas with the multiline version, one needs to understand how the dedent works with the final line.

The only thing that happens here is that the dedenter is _not broken_ w.r.t a final newline, there is nothing "extra" that one needs to understand. Avoiding to use a mechanism in any situation someone could imagine that it could break is not reasonable. That the test works as expected is enough context that nothing weird is going on.